game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/fire/SelectMainBattleCasualtiesTest.java

```diff
@@ -0,0 +1,526 @@
+package games.strategy.triplea.delegate.battle.steps.fire;
+
+import static games.strategy.triplea.Constants.EDIT_MODE;
+import static games.strategy.triplea.Constants.TRANSPORT_CASUALTIES_RESTRICTED;
+import static games.strategy.triplea.Constants.UNIT_ATTACHMENT_NAME;
+import static games.strategy.triplea.delegate.battle.FakeBattleState.givenBattleStateBuilder;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenAnyUnit;
+import static games.strategy.triplea.delegate.battle.steps.BattleStepsTest.givenUnitSeaTransport;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyCollection;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.hamcrest.MockitoHamcrest.argThat;
+
+import games.strategy.engine.data.GamePlayer;
+import games.strategy.engine.data.Unit;
+import games.strategy.engine.delegate.IDelegateBridge;
+import games.strategy.triplea.attachments.UnitAttachment;
+import games.strategy.triplea.delegate.DiceRoll;
+import games.strategy.triplea.delegate.battle.BattleState;
+import games.strategy.triplea.delegate.data.CasualtyDetails;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class SelectMainBattleCasualtiesTest {
+
+  @Mock IDelegateBridge delegateBridge;
+  @Mock GamePlayer gamePlayer;
+  BattleState battleState;
+
+  @BeforeEach
+  public void givenBattleState() {
+    battleState = givenBattleStateBuilder().build();
+  }
+
+  private DiceRoll givenDiceRollWithHits(final int hits) {
+    final DiceRoll diceRoll = mock(DiceRoll.class);
+    when(diceRoll.getHits()).thenReturn(hits);
+    return diceRoll;
+  }
+
+  @Test
+  void isEditMode() {
```

Comment:
nit, test method feels to be missing the implication of what happens when edit mode. A `@DisplayName` perhaps should become routine for any test method over 5 lines and particularly if there is more than one assert (think of it as a way to replace a javadoc on each test method, and each one should have a summary either in a direct title or display name to describe what we are testing).

Reply:
I've added a display name to all of the tests